sql/plan/call.go

```diff
@@ -163,6 +179,18 @@ func (c *Call) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
 	}, nil
 }
 
+// Database implements the sql.Databaser interface.
+func (c *Call) Database() sql.Database {
+	return c.db
+}
+
+// WithDatabase implements the sql.Databaser interface.
+func (c *Call) WithDatabase(db sql.Database) (sql.Node, error) {
+	nc := *c
+	nc.db = db
+	return &nc, nil
```

Comment:
Good job handling this subtle copy correctly! We've found some `With` functions that didn't make a new copy like they were supposed to and has caused weird errors.